File: leetcode/valid_palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string concat = "";

        // Getting all alphanumeric characters into a string
        for (int i = 0; i < s.length(); i++){
            if (iswalnum(s[i])){
                concat.push_back(s[i]);
            }
        }

        // Converting upper case to lower case
        transform(concat.begin(), concat.end(), concat.begin(), ::tolower);
        
        for (int k = 0; k < concat.length() / 2; k++){
            if (concat[k] != concat[concat.length() - 1 - k]){
                return false;
            }
        }
        return true;
    }
};

File: neetcode/two_pointers/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        fls = "".join(ch.lower() for ch in s if ch.isalnum())
        for i in range(int(len(fls) / 2)):
            if fls[i] != fls[len(fls) - 1 - i]:
                return False
        return True
